processor/src/trace/tests/hasher.rs

```diff
@@ -0,0 +1,163 @@
+use super::{
+    super::{Trace, NUM_RAND_ROWS},
+    build_trace_from_ops_with_inputs, rand_array, Felt, LookupTableRow, Operation, ProgramInputs,
+    Word, ONE, ZERO,
+};
+use crate::hasher::SiblingTableRow;
+use vm_core::{
+    AdviceSet, FieldElement, StarkField, AUX_TRACE_RAND_ELEMENTS, HASHER_AUX_TRACE_OFFSET,
+};
+
+// CONSTANTS
+// ================================================================================================
+
+const P1_COL_IDX: usize = HASHER_AUX_TRACE_OFFSET;
+
+// SIBLING TABLE TESTS
+// ================================================================================================
+
+#[test]
+#[allow(clippy::needless_range_loop)]
+fn hasher_p1_mp_verify() {
+    let tree = build_merkle_tree();
+    let node = tree.get_node(3, 1).unwrap();
+
+    // build program inputs
+    let mut init_stack = vec![3, 1];
+    append_word(node, &mut init_stack);
+    append_word(tree.root(), &mut init_stack);
```

Comment:
nit: I would probably reverse the order of parameters here. It feels more logically consistent to have the thing we're appending to come first, so I think it would be faster to process what's going on.

Reply:
Agreed! Changed.